### PR TITLE
CB-13493 Fix Datahub HA repair, remove primary gateway change

### PR DESCRIPTION
**Pull request by lacikaaa**
- Primary gateway change messes up the salt roles and PGW on CB side. As CM is non HA, PGW doesn't make any sense, similarly to DL.
- Fixed downscale when the instance is not returned by AWS as it's deleted long time ago. In this case no termination request is sent for the instance.
- When fetching all `GatewayConfig` only the reachables are returned to avoid connection issues. If there is no gateway instance reachable an error is raised.

See detailed description in the commit message.

### CODE REVIEW

**Review by pdarvasi — CHANGES_REQUESTED** [review]

File: cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsDownscaleService.java
Comment (lacikaaa) [reply cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsDownscaleService.java:164]: right

File: core-model/src/main/java/com/sequenceiq/cloudbreak/domain/stack/Stack.java
Comment (lacikaaa) [reply core-model/src/main/java/com/sequenceiq/cloudbreak/domain/stack/Stack.java:592]: no, it's not in scope for this PR

**Review by sodre90 — APPROVED** [review]

**Comment by pdarvasi** [comment]
@sodre90 next time if I've made the effort to do a review, pls wait for my approval before you'd merge, thank you!

File: core-model/src/main/java/com/sequenceiq/cloudbreak/domain/stack/Stack.java
Comment (pdarvasi) [reply core-model/src/main/java/com/sequenceiq/cloudbreak/domain/stack/Stack.java:592]: ok, are you confident if we need both methods?

Comment (lacikaaa) [reply core-model/src/main/java/com/sequenceiq/cloudbreak/domain/stack/Stack.java:592]: yes, I'm pretty sure sometimes we need all GWs which are not terminated, while when creating connectors/clients it doesn't make sense to try to reach a GW which is stopped

**Pull request merged**
